plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/ConfigureDependentFrameworksListener.java

```diff
@@ -0,0 +1,212 @@
+				String tlsVerifyString = adb.env.get("DOCKER_TLS_VERIFY");
+				boolean tlsVerify = (Integer.parseInt(tlsVerifyString) != 0);
+				// For some reason, the connection fails if i set this. 
+				// Not sure why the disconnect between what adbinfo tells me to use and what works upstream
+//				if( tlsVerify ) {
+//					String tlsCertPath = adb.env.get("DOCKER_CERT_PATH");
+//					tcpConnectionBuilder.tcpCertPath(tlsCertPath);
+//				}
+				DockerConnection con = tcpConnectionBuilder.build();
+				IDockerConnection[] other = mgr.getConnections();
+				mgr.addConnection(con);
+			}
+		} catch(DockerException de) {
+			de.printStackTrace();
+		}
+	}
+	private void configureOpenshift(ADBInfo adb) {
+		Collection<IConnection> connections = ConnectionsRegistrySingleton.getInstance().getAll();
+		Iterator<IConnection> it = connections.iterator();
+		String soughtHost = adb.openshiftHost + ":" + adb.openshiftPort;
+		boolean found = false;
+		while(it.hasNext() && !found) {
+			IConnection c = it.next();
+			if( c.getType() == ConnectionType.Kubernetes) {
+				String host = c.getHost();
+				if( host.equals(soughtHost)) {
+					found = true;
+				}
+			}
+		}
+		
+		if( !found ) {
+			ConnectionsFactoryTracker connectionsFactory = new ConnectionsFactoryTracker();
+			connectionsFactory.open();
+			IConnectionFactory factory = connectionsFactory.getById(IConnectionsFactory.CONNECTIONFACTORY_OPENSHIFT_ID);
+			IConnection con = factory.create(soughtHost);
+			((Connection)con).setAuthScheme("Basic");
+			((Connection)con).setUsername("test-admin");
+			((Connection)con).setPassword("password");
+			ConnectionsRegistrySingleton.getInstance().add(con);
+		}
+	}
+	
+	private ADBInfo loadADBInfo(IServer server) {
+		try {
+			ILaunchConfigurationWorkingCopy wc = new CDKLaunchConfigUtility().
+					createExternalToolsLaunchConfig(server, "adbinfo", server.getName() + " adbinfo");
+			ILaunch launch2 = wc.launch("run", new NullProgressMonitor());
+			final IProcess[] processes = launch2.getProcesses();
+			IStreamMonitor mon = processes[0].getStreamsProxy().getOutputStreamMonitor();
+			final ArrayList<String> lines = new ArrayList<String>();
+			mon.addListener(new IStreamListener() {
+				public void streamAppended(String text, IStreamMonitor monitor) {
+					lines.add(text);
+				}
+			});
+			
+			while( !processes[0].isTerminated()) {
+				try {
+					Thread.sleep(500);
+				} catch(InterruptedException ie) {
+					// ignore
+				}
+			}
+			
+			HashMap<String,String> env = new HashMap<String,String>();
+			Iterator<String> lineIterator = lines.iterator();
+			while(lineIterator.hasNext()) {
+				String oneAppend = lineIterator.next();
+				String[] allAppends = oneAppend.split("\n");
+				for( int i = 0; i < allAppends.length; i++ ) {
+					if( allAppends[i].trim().startsWith("export ")) {
+						int eq = allAppends[i].indexOf("=");
+						if( eq != -1 ) {
+							String k = allAppends[i].substring("export ".length(), eq);
```

Comment:
Check above calls trim() but key extraction does not. Both should either to call trim() or not do it.